### ISSUE TITLE
Center editorial accidentals on note

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#396]
Editorial accidentals are placed so that the left side of the accidental is centered on the notehead.  Rather the accidental should be centered on the center of the notehead.  This applies to all accidentals:

<img width="684" alt="screen shot 2016-12-24 at 20 55 49" src="https://cloud.githubusercontent.com/assets/3487289/21469775/6daf3bb6-ca1b-11e6-8446-298a1d18bffe.png">

Also I had to use `accid@accid="ss"` for the double sharp.  Should this not be `accid@accid="x"`?
(`@accid.ges="ss"` is a gestural double sharp).

Also, the apparent height of the accidentals above the notes does not look equal, which would probably be better.  Either the sharp is too low or the flat is too high, for example.

Test MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-12-24 20:55:14</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-8b82cb1-dirty</p>
            </projectDesc>
        </encodingDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="8" token="!!!RDF**kern: i = editorial accidental" xml:id="loc8">
                    <frameInfo>
                        <startTime float="16" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>i = editorial accidental</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000027977328">
                        <staffGrp xml:id="m-000000028778056">
                            <staffDef xml:id="staffdef-000000129725177" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000076944697">
                        <measure xml:id="measure-L2" n="1" right="end">
                            <staff xml:id="staff-L2F1" n="1">
                                <layer xml:id="layer-L2F1" n="1">
                                    <note xml:id="note-L3F1" dur="1" oct="5" pname="f">
                                        <accid xml:id="accid-000000113038940" accid="s" func="edit" />
                                    </note>
                                    <note xml:id="note-L4F1" dur="1" oct="5" pname="f">
                                        <accid xml:id="accid-000000162539045" accid="f" func="edit" />
                                    </note>
                                    <note xml:id="note-L5F1" dur="1" oct="5" pname="f">
                                        <accid xml:id="accid-000000208946960" accid="n" func="edit" />
                                    </note>
                                    <note xml:id="note-L6F1" dur="1" oct="5" pname="f">
                                        <accid xml:id="accid-000000207644959" accid="ss" func="edit" />
                                    </note>
                                    <note xml:id="note-L7F1" dur="1" oct="5" pname="f">
                                        <accid xml:id="accid-000000179518178" accid="ff" func="edit" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by donbyrd** [comment]
@craigsapp said "Either the sharp is too low or the flat is too high..."  The sharp is certainly too low! Likewise the natural. If they don't actually touch the note head, they come awfully close. He also said "[T]he apparent height of the accidentals above the notes does not look equal, which would probably be better." I suspect that if they were all moved up by, say, a linespace, the apparent heights would look equal enough, but just moving the sharp and natural up might be better.

**Comment by craigsapp** [comment]
Here are the horizontal justifications for editorial accidentals in SCORE:

<img width="1125" alt="editpos" src="https://cloud.githubusercontent.com/assets/3487289/21486283/825c1498-cb66-11e6-867a-d0d4266341ae.png">

The red line marks the apex of the curvature of the notehead.  The green lines are the left/right position of the accidental on the note.  The orange horizontal line is 1/2 the width of the flat acciental.

It looks like the justification rule in SCORE is:  place the editorial flat centered at the apex of the notehead.  For sharps and naturals, they are  right justified to the position of the flat accidental's right edge.

**Comment by craigsapp** [comment]
Here are the default vertical justifications for editorial accidentals in SCORE:

<img width="1125" alt="editheight" src="https://cloud.githubusercontent.com/assets/3487289/21486350/164d92a2-cb68-11e6-991e-f2f9ca887a62.png">

Flats are noticeably higher than naturals or sharps.  This is due to the placement being exactly a diatonic 4th above the notes in the example. (I would prefer that the flat be dropped to the green line to match the bottom boundary of the other two accidentals).  Current verovio behavior is to place the 

Example vertical positions for a range of pitches/stem directions:

<img width="1168" alt="screen shot 2016-12-26 at 12 42 00" src="https://cloud.githubusercontent.com/assets/3487289/21486399/c486d43c-cb68-11e6-8464-9fc6c4cfdd32.png">

**Comment by craigsapp** [comment]
Here is an example of the sizes of the editorial accidentals in SCORE:

<img width="1098" alt="screen shot 2016-12-26 at 12 51 53" src="https://cloud.githubusercontent.com/assets/3487289/21486470/2abde2a8-cb6a-11e6-9f40-580a1904bfd8.png">

The editorial accidentals are exactly 65% of the size of a regular accidental.  This is slightly larger than cue-sized accidentals.  Cue-sized notes are 60% of normal size as illustrated below (and cue sized editorial accidentals are 39% of full size).

<img width="544" alt="screen shot 2016-12-26 at 13 01 13" src="https://cloud.githubusercontent.com/assets/3487289/21486580/9414345e-cb6b-11e6-895d-168e6305079f.png">

**Comment by donbyrd** [comment]
Elaine Gould doesn't agree on the size of cue-size notes. In _Behind Bars_, she says a cue note "is" 3/4 of a full-size note (p. 125), and cue notation "is about" 3/4 the size of full notation (p. 569).

**Comment by craigsapp** [comment]
Nobody is perfect :-). But then, 65% is "about" 75%.  What does she say about editorial accidental sizes?

**Comment by donbyrd** [comment]
She says editorial accidentals can be "given as small-size accidentals above their notes"  :-) . (She also discusses putting them before notes in brackets, in which case they'd presumably be full-size.)

**Comment by craigsapp** [comment]
>(She also discusses putting them before notes in brackets, in which case they'd presumably be full-size.)

Yes.  I won't be needing that style, but would be useful to have available with verovio.  How would they be encoded?  Perhaps presence of `@enclose` would trigger that?  But what if someone wanted an editorial accidental above the note, but also in brackets?

**Comment by craigsapp** [comment]
For interest,  here is a sample of a note at size 100%, 60% and 75%:

<img width="986" alt="screen shot 2016-12-26 at 23 02 01" src="https://cloud.githubusercontent.com/assets/3487289/21493251/697275d6-cbbf-11e6-99d9-050108f6d44b.png">

SCORE actually calls 60% the "grace-note size", not cue size.  I don't see a default size for cue notes in SCORE.

**Comment by donbyrd** [comment]
Ah-hah! Gould also says (p. 125) "the grace note is slightly smaller than a cue note."

I also looked at Ted Ross' classic _The Art of Music Engraving and Processing_, but he gives rules in terms of "rastrals", the old engraver's standard sizes, rules that don't make much sense now: for sizes 1, 2, and 3, cue notes should be in size 4; for sizes 4, 5, and 6, they should be in size 7 (I think -- I don't have it in front of me at the moment).

**Comment by craigsapp** [comment]
Somehow editorial accidental horizontal placement was altered recently.  The editorial accidentals are now center on the left edge of a notehead.  Instead they should be centered on the center of a notehead.

Example of the current behavior:

<img width="764" alt="screen shot 2017-04-18 at 5 01 12 pm" src="https://cloud.githubusercontent.com/assets/3487289/25157877/fc6b2648-2458-11e7-937a-2d4534d9df7a.png">

Text MEI data:


```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-04-18T17:01:03" version="0.9.14-dev-d2d89fc">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000955244576">
                        <staffGrp xml:id="staffgrp-0000001695080009">
                            <staffDef xml:id="staffdef-0000002026256741" n="1" clef.shape="G" clef.line="2" key.sig="0" meter.count="2" meter.unit="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001734885525">
                        <app xml:id="app-0000002049667859">
                            <lem xml:id="lem-0000001693939643" />
                            <rdg xml:id="rdg-0000001622633240" label="original-clef">
                                <scoreDef xml:id="scoredef-0000001574768912">
                                    <staffGrp xml:id="staffgrp-0000002108295573">
                                        <staffDef xml:id="staffdef-0000001383614265" n="1" clef.shape="C" clef.line="1" />
                                    </staffGrp>
                                </scoreDef>
                            </rdg>
                        </app>
                        <measure xml:id="measure-L11" n="5" type="m-5">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L12F1" dur="4" oct="5" pname="d">
                                        <accid xml:id="accid-L12F1" accid.ges="n" />
                                    </note>
                                    <beam xml:id="beam-L13F1-L14F1">
                                        <note xml:id="note-L13F1" dur="8" oct="5" pname="c">
                                            <accid xml:id="accid-L13F1" accid="s" func="edit" />
                                        </note>
                                        <note xml:id="note-L14F1" dur="8" oct="4" pname="b">
                                            <accid xml:id="accid-L14F1" accid.ges="n" />
                                        </note>
                                    </beam>
                                    <note xml:id="note-L15F1" dur="2" oct="5" pname="c">
                                        <accid xml:id="accid-L15F1" accid="s" func="caution" />
                                    </note>
                                    <note xml:id="note-L16F1" dur="4" oct="5" pname="d">
                                        <accid xml:id="accid-L16F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L17F1" dur="4" oct="5" pname="d">
                                        <accid xml:id="accid-L17F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L18F1" dur="4" oct="4" pname="b">
                                        <accid xml:id="accid-L18F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L19F1" dur="4" oct="5" pname="c">
                                        <accid xml:id="accid-L19F1" accid="n" func="edit" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
I forgot to drop some code. Fixed ae687bd7486d0246788be0c0dc3a75558386699f

![image](https://cloud.githubusercontent.com/assets/689412/25200449/a5d2d208-254e-11e7-9bb0-a27bdefffa85.png)
